### PR TITLE
Fixed cyclic import errors

### PR DESCRIPTION
**Pull request by mikand**
This PR addresses issue #405 .

I created a test that invokes the python interpreter and checks that each pysmt module (except the testing ones and the solvers) can be imported as a standalone (i.e. without previously importing something else).

In order for this test to work I had to move some imports of the environment module inside some functions, but I think this trade-off is acceptable.

### CODE REVIEW

**Comment by marcogario** [comment]
I am not sure that the issue is particularly critical, and I fear that the fix is polluting other functions without a real gain.

The circular dependency mentioned in #405 was introduced after we made the type information become part of the environment. The import of the environment occurs at Line 500 and impacts only 4 types. This was done for backwards compatibility, but if we want to get rid of this, then we should move those into shortcuts.py (indeed, if you look at the implementation they are shortcuts) [1]. 

The other point is whether there are other cyclic imports and whether we want to deal with them. Note that this fix does not remove the dependency itself, but only allows the import to complete. Arguably, one could ask whether the dependency is necessary in the first place or could be refactored away.

The one that I do not like is the one on ``fnode.py:_env`` this function gets called everytime we perform an infix notation operation. Nevertheless, this is needed since we want to access the global environment.

In summary, I would accept the change in ``oracle.py`` and ``fnode.py`` if we do not find another solution. I would prefer to find a better solution for ``typing.py``.

 [1] Since we are waiting to release 0.7.0 this is the moment to break backwards compatibility!

**Comment by mikand** [comment]
@marcogario  You are right.

Check out the changes now, there is a much simpler solution to import ```pysmt.enviornment``` without causing cyclic dependencies, that is importing ```pysmt```. This will make all the modules (not the packages) within the ```pysmt``` package immediately available without executing them.

Do you think this solution is acceptable?

**Comment by marcogario** [comment]
Not 100% sure of why this works, but it looks much better. 

To be sure, I tried generating and installing the pkg and it also works after installing it via pip.
